treeherder/etl/tasks/pulse_tasks.py

```diff
@@ -16,7 +17,10 @@ def store_pulse_jobs(job_list, exchange, routing_key):
     newrelic.agent.add_custom_parameter("exchange", exchange)
     newrelic.agent.add_custom_parameter("routing_key", routing_key)
 
-    JobLoader().process_job_list(job_list)
+    try:
+        JobLoader().process_job_list(job_list)
+    except MissingResultsetException as exc:
+        store_pulse_jobs.retry(exc=exc)
```

Comment:
Yeah, it's true that it isn't going through ``retryable_task`` with this.  Though the normal retry still honors the ``max_retries`` setting.  I verified that with a new test I just added.  So it quits and just throws the exception.  The default retry interval of 3 minutes is really probably just fine, imo.  Though, happy to do something different if you feel strongly different.  :)

James and I actually discussed your idea about special-casing ``MissingResultsetException`` or adding the ability to specify a special case exception on ``retryable_task``.  We could certainly do that, though I opted for this because it seemed simpler.  But, again, happy to discuss and go this route if it seems significantly better.  :)